pandas/tests/groupby/test_groupby.py

```diff
@@ -195,6 +195,42 @@ def f(grp):
     tm.assert_series_equal(result, e)
 
 
+@pytest.mark.parametrize(
+    "obj",
+    [
+        tm.SubclassedDataFrame({"A": np.arange(0, 10)}),
+        tm.SubclassedSeries(np.arange(0, 10), name="A"),
+    ],
+)
+def test_groupby_preserves_subclass(obj, groupby_func):
```

Comment:
does this cover *all* of the changes? e.g. I see some that are in non-groupby files, please add tests for those as well.

Reply:
To my knowledge, it does hit almost all of the changes in other files (I'll add any needed tests after combing through the codecov report).  For example:
- the changes in `frame.py` end up being hit by the `groupby.idxmin()` and `groupby.idxmax()` tests
- the changes to `concat.py` end up being hit by the `groupby.transform()` operations via `DataFrameGroupBy.__transform_general()`

I will add more targeted tests though to make sure other ways of calling those functions still preserve subclassing.